topdown/regex.go

```diff
@@ -215,4 +241,5 @@ func init() {
 	RegisterFunctionalBuiltin4(ast.RegexTemplateMatch.Name, builtinRegexMatchTemplate)
 	RegisterFunctionalBuiltin3(ast.RegexFind.Name, builtinRegexFind)
 	RegisterFunctionalBuiltin3(ast.RegexFindAllStringSubmatch.Name, builtinRegexFindAllStringSubmatch)
+	RegisterFunctionalBuiltin3(ast.RegexReplace.Name, builtinRegexReplace)
```

Comment:
This register function is deprecated, please use [`RegisterBuiltinFunc`](https://sourcegraph.com/github.com/open-policy-agent/opa/-/blob/topdown/builtins.go?L86:4&subtree=true), and adapt the function signature accordingly... 😳  I'm sorry it's hard to avoid this trap, we should be updating all the register calls eventually...

Reply:
Indeed! I had a branch up on my fork at one point for refactoring away the older style, at least within the standard library of builtins. I'll have to see if I can revive it.

Reply:
thanks for the review! I've transformed the registration to be using `RegisterBuiltinFunc()`. btw (just an idea) given all this transformation is straightforward and low-hanging fruits for new comers, it might be beneficial to mark those as good-first-issue

```shell
egrep -rnw "RegisterFunctionalBuiltin2|RegisterFunctionalBuiltin3|RegisterFunctionalBuiltin4" ./ 2>/dev/null | wc -l
51
```